cli_044_global_land_temperature/contents/src/__init__.py

```diff
@@ -0,0 +1,362 @@
+    '''
+    # get dates that are already in the table along with the new dates that are already processed
+    seen_ids = existing_ids + list(new_data.keys())
+    # if the current new date is not in the existing table and has not processed yet, add it to the dictionary of new data
+    if newUID not in seen_ids:
+        new_data[newUID] = newValues
+        logging.debug("Adding {} data to table".format(newUID))
+    else:
+        logging.debug("{} data already in table".format(newUID))
+
+    return(new_data)   
+
+def processData(url, existing_ids, date_format='%Y-%m-%d %H:%M:%S'):
+    '''
+    Fetch, process and upload new data
+    INPUT   url: url where you can find the download link for the source data (string)
+            existing_ids: list of date IDs that we already have in our Carto table (list of strings)
+            date_format: format of dates in Carto table (string)
+    RETURN  num_new: number of rows of new data sent to Carto table (integer)
+    '''
+    # initialize variable to store number of new rows sent to Carto
+    num_new = 0
+    # Get the link from source url for which we want to download data
+    resource_location = fetchDataFileName(url)
+    # get the data from source as a list of strings, with each string holding one line from the source data file
+    res_rows = tryRetrieveData(url, resource_location)
+    # create an empty dictionary to store new data (data that's not already in our Carto table)
+    new_data = {}
+    # remove headers by deleting first five rows
+    # !!!I TRIED TO FIND MORE POLISHED WAY TO ACHIEVE THIS BUT NONE OF THEM WERE WORKING CORRECTLY!!!
+    res_rows = res_rows[5:]
+
+    # go through each line of content retrieved from source
+    for row in res_rows:
+        # split line by space to get each columns as separate elements
+        row = row.split()
+        logging.debug("Processing row: {}".format(row))
+        # get year from the first column
+        year = int(row[0])
+        # construct datetime object using year and first day of January 
+        date_obj = datetime.datetime(year, 1, 1)
+        # convert datetime object to string formatted according to date_pattern
+        date = date_obj.strftime(date_format)
+        # get unsmoothed temperature change by accessing the second column 
+        no_smoothing = float(row[1])
+        # get five year Lowess smoothed temperature change by accessing the last column
+        lowess_smoothing = float(row[-1])
+        # store all the variables into a list
+        values = [date, no_smoothing, lowess_smoothing]
+        # For new date, check whether this is already in our table. 
+        # If not, add it to the queue for processing
+        new_data = insertIfNew(date, values, existing_ids, new_data)
+
+    # if we have found new data to process
+    if len(new_data):
+        num_new += len(new_data)
```

Comment:
since this isn't being used in a loop and recursively adding new rows each time, we don't need this to be +=...it can really just be =
```suggestion
        num_new = len(new_data)
```